avocado/core/test.py

```diff
@@ -941,12 +953,19 @@ def _setup_environment_variables(self):
         if self.datadir is not None:
             os.environ['AVOCADO_TEST_DATADIR'] = self.datadir
         os.environ['AVOCADO_TEST_WORKDIR'] = self.workdir
-        os.environ['AVOCADO_TEST_SRCDIR'] = self.srcdir
         os.environ['AVOCADO_TEST_LOGDIR'] = self.logdir
         os.environ['AVOCADO_TEST_LOGFILE'] = self.logfile
         os.environ['AVOCADO_TEST_OUTPUTDIR'] = self.outputdir
         if self.__sysinfo_enabled:
             os.environ['AVOCADO_TEST_SYSINFODIR'] = self.__sysinfodir
+        # srcdir is deprecated and will cause a test warning when
+        # accessed.  It seems unfair to return a warning for all
+        # tests because Avocado itself will access that property.
+        # this is a hack to be removed when srcdir is also removed
+        # for good.
+        self.__srcdir_internal_access = True
+        os.environ['AVOCADO_TEST_SRCDIR'] = self.srcdir
+        self.__srcdir_internal_access = False
```

Comment:
Oh, this is new as well. Well, to be honest I'd prefer some other solution like additional argument or `_srcdir` method to just get the value. But it'll be removed quite soon so let's keep it this way. (imagine concurrency issues)